types/p-defer/index.d.ts

```diff
@@ -1,16 +1,17 @@
 // Type definitions for p-defer 1.0
 // Project: https://github.com/sindresorhus/p-defer
 // Definitions by: Sam Verschueren <https://github.com/SamVerschueren>
+//                 BendingBender <https://github.com/BendingBender>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-declare namespace pDefer {
-	interface DeferredPromise<T> {
-		resolve<U>(value?: U | PromiseLike<U>): Promise<U>;
-		reject(reason: any): Promise<never>;
-		promise: Promise<T>;
-	}
-}
+export = pDefer;
 
 declare function pDefer<T>(): pDefer.DeferredPromise<T>;
 
-export = pDefer;
+declare namespace pDefer {
+    interface DeferredPromise<T> {
+        resolve(value?: T | PromiseLike<T>): void;
```

Comment:
I think `resolve` and `reject` usually return promises.  Does this library work differently?

Reply:
No, reolve and reject pretty much always return undefined, this lib is no exception.
See:
https://www.ecma-international.org/ecma-262/6.0/#sec-promise-resolve-functions

From lib.es2015.promise.d.ts:
```
    new <T>(executor: (resolve: (value?: T | PromiseLike<T>) => void, reject: (reason?: any) => void) => void): Promise<T>;

```

Reply:
I'm convinced :smile: